lib/internal/Magento/Framework/Image/Adapter/ImageMagick.php

```diff
@@ -499,4 +480,84 @@ protected function _getImagickPixelObject($color = null)
     {
         return new \ImagickPixel($color);
     }
+
+    /**
+     * Resizes watermark to desired size, when it is not stretched
+     *
+     * @param \Imagick $watermark
+     */
+    private function resizeWatermerk(\Imagick $watermark): void
+    {
+        if ($this->getWatermarkWidth() &&
+            $this->getWatermarkHeight() &&
+            $this->getWatermarkPosition() != self::POSITION_STRETCH
+        ) {
+            $watermark->resizeImage(
+                $this->getWatermarkWidth(),
+                $this->getWatermarkHeight(),
+                \Imagick::FILTER_CUBIC,
+                self::BLUR_FACTOR
+            );
+        }
+    }
+
+    /**
+     * Keeps transparenty if watermark is transparent
+     * @param \Imagick $watermark
+     */
+    private function handleWatermarkAlphaChannel(\Imagick $watermark): void
+    {
+        if (method_exists($watermark, 'getImageAlphaChannel')) {
+            // available from imagick 6.4.0
+            if ($watermark->getImageAlphaChannel() == 0) {
+                $watermark->setImageAlphaChannel(\Imagick::ALPHACHANNEL_OPAQUE);
+            }
+        }
+    }
+
+    /**
+     * Add tiled watermark at starting given X,Y position
+     * @param $positionX
+     * @param $positionY
+     * @param \Imagick $watermark
+     * @param bool $compositeChannels
+     */
+    private function addTiledWatermark($positionX, $positionY, \Imagick $watermark, bool $compositeChannels): void
+    {
+        $offsetX = $positionX;
+        $offsetY = $positionY;
+        while ($offsetY <= $this->_imageSrcHeight + $watermark->getImageHeight()) {
+            while ($offsetX <= $this->_imageSrcWidth + $watermark->getImageWidth()) {
+                $this->_imageHandler->compositeImage(
+                    $watermark,
+                    \Imagick::COMPOSITE_OVER,
+                    $offsetX,
+                    $offsetY,
+                    $compositeChannels
+                );
+                $offsetX += $watermark->getImageWidth();
+            }
+            $offsetX = $positionX;
+            $offsetY += $watermark->getImageHeight();
+        }
+    }
+
+    /**
+     * Add watermark at given X,Y position
+     *
+     * @param $positionX
+     * @param $positionY
+     * @param \Imagick $watermark
```

Comment:
I'd import the `\Imagick` class.

Reply:
What is the benefit of such import?

Reply:
The imports are consistent.

Reply:
I followed this suggestion and phpstan raged. Reversing